### ISSUE TITLE
How to generate Figure 2(b)?

### ISSUE THREAD

**Issue by waychin-weiqin** [gangweiX/IGEV#29]
Hi, 

Thank you for sharing this amazing work! I am wondering how did you generate Figure 2b (shown in screenshot below) in the paper? Did you first generate the all-pairs correlations using the following function? If so, how did you go from there to disparity map? 

 https://github.com/gangweiX/IGEV/blob/ea4d55bca430971884fd3e4515dd868cdf2abf88/IGEV-Stereo/core/geometry.py#L62

![Screenshot 2023-07-26 at 3 53 38 pm](https://github.com/gangweiX/IGEV/assets/33308266/f1eefc75-a039-4dfa-8e75-2863bde077e7)

Thank you in advance for your help with my question :)

**Comment by gangweiX** [comment]
By function corr(fmap1, fmap2), I can get correlation [B, H, W1, W2]. along the last dim, apply softmax function to compute disparity.

**Comment by waychin-weiqin** [comment]
Thank you so much for the prompt reply! Can you clarify if to apply softmax or soft-argmax to the last dim of the correlation volume? 

Thank you again :)

**Comment by gangweiX** [comment]
pseudo code：
# CV : [B, H, W1, W2]
prob = F.softmax(CV, dim=-1)
disp = torch.arange(0, W2, dtype=prob.dtype, device=prob.device)
disp  = disp .view(1, 1, 1, W2)
disp = torch.sum(prob * disp , dim=-1)
coords = torch.arange(W2).float().to(disp.device).reshape(1,1,W2).repeat(B, H, 1)
final_disp = coords - disp

I wrote it by hand, without debugging, maybe it has some mistakes, but the process is like this
